packages/autocertifier/src/AutoCertifier.ts

```diff
@@ -0,0 +1,195 @@
+        if (!databaseFilePath) {
+            throw new Error('AUTOCERTIFIER_DATABASE_FILE_PATH environment variable is not set')
+        }
+
+        const accountPrivateKeyPath = process.env['AUTOCERTIFIER_ACCOUNT_PRIVATE_KEY_PATH']
+        if (!accountPrivateKeyPath) {
+            throw new Error('AUTOCERTIFIER_ACCOUNT_PRIVATE_KEY_PATH environment variable is not set')
+        }
+
+        const acmeDirectoryUrl = process.env['AUTOCERTIFIER_ACME_DIRECTORY_URL']
+        if (!acmeDirectoryUrl) {
+            throw new Error('AUTOCERTIFIER_ACME_DIRECTORY_URL environment variable is not set')
+        }
+
+        const hmacKid = process.env['AUTOCERTIFIER_HMAC_KID']
+        if (!hmacKid) {
+            throw new Error('AUTOCERTIFIER_HMAC_KID environment variable is not set')
+        }
+
+        const hmacKey = process.env['AUTOCERTIFIER_HMAC_KEY']
+        if (!hmacKey) {
+            throw new Error('AUTOCERTIFIER_HMAC_KEY environment variable is not set')
+        }
+
+        const restServerCaCertPath = process.env['AUTOCERTIFIER_REST_SERVER_CA_CERT_PATH']
+        if (!restServerCaCertPath) {
+            throw new Error('AUTOCERTIFIER_REST_SERVER_CA__CERT_PATH environment variable is not set')
+        }
+
+        const restServerCaKeyPath = process.env['AUTOCERTIFIER_REST_SERVER_CA_KEY_PATH']
+        if (!restServerCaKeyPath) {
+            throw new Error('AUTOCERTIFIER_REST_SERVER_CA_KEY_PATH environment variable is not set')
+        }
+
+        const restServerCertPath = process.env['AUTOCERTIFIER_REST_SERVER_CERT_PATH']
+        if (!restServerCertPath) {
+            throw new Error('AUTOCERTIFIER_REST_SERVER_CERT_PATH environment variable is not set')
+        }
+
+        const restServerKeyPath = process.env['AUTOCERTIFIER_REST_SERVER_KEY_PATH']
+        if (!restServerKeyPath) {
+            throw new Error('AUTOCERTIFIER_REST_SERVER_KEY_PATH environment variable is not set')
+        }
+
+        this.database = new Database(databaseFilePath)
+        await this.database.start()
+        logger.info('database is running on file ' + databaseFilePath)
+
+        this.dnsServer = new DnsServer(this.domainName, ownHostName, dnsServerPort,
+            ownIpAddress, this.database)
+        await this.dnsServer.start()
+        logger.info('dns server is running for domain ' + this.domainName + ' on port ' + dnsServerPort)
+
+        this.certificateCreator = new CertificateCreator(acmeDirectoryUrl, hmacKid, hmacKey,
+            accountPrivateKeyPath, this)
+        logger.info('certificate creator is running')
+
+        this.restServer = new RestServer(ownHostName + '.' + this.domainName, ownIpAddress, restServerPort, restServerCaCertPath, restServerCaKeyPath,
+            restServerCertPath, restServerKeyPath, this)
+        await this.restServer.start()
+        logger.info('rest server is running on port ' + restServerPort)
+    }
+
+    public async stop(): Promise<void> {
+        if (this.restServer) {
```

Comment:
We can maybe assume that stop isn't called before start completes, and therefore remove these if checks